### ISSUE TITLE
SelfTest fails

### ISSUE THREAD

**Issue by awakeford** [Legrandin/pycryptodome#364]
Using pycryptodomex-3.9.4

>>> import Cryptodome.SelfTest
>>> Cryptodome.SelfTest.run()

Fails with:

Skipping AESNI tests
Skipping test of PCLMULDQD in AES GCM
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/proj/digig5_1_sjc2/devel/wakeford/crypotn100_tb_b/trunk/tbcs/python_lib/lib/python3.5/site-packages/Cryptodome/SelfTest/__init__.py", line 73, in run
    result = runner.run(suite)
  File "/usr/opt/pmc/python/python-3.5.2.x86_64/lib/python3.5/unittest/runner.py", line 185, in run
    self.stream.writeln(result.separator2)
  File "/usr/opt/pmc/python/python-3.5.2.x86_64/lib/python3.5/unittest/runner.py", line 25, in writeln
    self.write(arg)
TypeError: a bytes-like object is required, not 'str'

**Comment by Legorooj** [comment]
Which version are you running? OS, python, pycryptodome...

**Comment by awakeford** [comment]
Thanks for looking into it.

**Comment by texadactyl** [comment]
Suggestion: put in a test for Python 2 (deprecated):
```
REQUIRED_MAJOR = 3
REQUIRED_MINOR = 7

if sys.version_info.major < REQUIRED_MAJOR \
or sys.version_info.minor < REQUIRED_MINOR:
    print("This script requires Python {}.{} or higher!"
          .format(REQUIRED_MAJOR, REQUIRED_MINOR))
    print("You are using Python {}.{}."
          .format(sys.version_info.major, sys.version_info.minor))
    sys.exit(86)

```

**Comment by Legrandin** [comment]
Fixed with https://github.com/Legrandin/pycryptodome/commit/60ced09401fb229fdbcda0eb9a43be52ebe08a26

**Comment by Legrandin** [comment]
Fixed in version 3.10.3.
